spec/migration_spec.rb

```diff
@@ -1,24 +1,83 @@
+
+      allow(described_class.anonymous_class_with_id_regions.connection).to receive(:select_value).and_wrap_original do |m, arg|
+        if arg == "SELECT relname FROM pg_class WHERE relkind = 'S' LIMIT 1"
+          # HACK the order so that we get the most recently created
+          m.call("SELECT relname FROM pg_class WHERE relkind = 'S' ORDER BY oid DESC LIMIT 1")
+        else
+          m.call(arg)
+        end
+      end
+
+      suppress_migration_messages do
+        Class.new(ActiveRecord::Migration[version]) do
+          def change
+            create_table :testing_correct_sequence
+          end
+        end.migrate(:up)
       end
+
+      expect(TestRecord.id_to_region(ActiveRecord::Base.connection.select_value("SELECT last_value FROM testing_correct_sequence_id_seq"))).to eq(region_number_before)
     end
 
-    Class.new(ActiveRecord::Migration[5.0]) do
-      def change
-        create_table :testing_correct_sequence
+    context "Tests create_table for #{version} migrations" do
+      let(:klass) { Class.new(ActiveRecord::Base) }
+
+      it "id => false, creates no id column" do
+        suppress_migration_messages do
+          Class.new(ActiveRecord::Migration[version]) do
+            def change
+              create_table :testing_create_table_id_false, :id => false do |t|
+                t.string "name"
+              end
+            end
+          end.migrate(:up)
+        end
+
+        klass.table_name = :testing_create_table_id_false
+        expect(klass.attribute_names.sort).to eq(%w(name))
+      end
+
+      it "implicit id, creates bigserial id" do
+        suppress_migration_messages do
+          Class.new(ActiveRecord::Migration[version]) do
+            def change
+              create_table :testing_create_table_implicit_id do |t|
+                t.string "name"
+              end
+            end
+          end.migrate(:up)
+        end
+
+        klass.table_name = :testing_create_table_implicit_id
+        expect(klass.attribute_names.sort).to eq(%w(id name))
+
+        klass.create!(:name => "abc")
+        expect(klass.first.id).to be > TestRecord::DEFAULT_RAILS_SEQUENCE_FACTOR
       end
-    end.migrate(:up)
 
-    expect(TestRecord.id_to_region(ActiveRecord::Base.connection.select_value("SELECT last_value FROM testing_correct_sequence_id_seq"))).to eq(region_number_before)
+      it "id => integer, creates bigserial id" do
+        suppress_migration_messages do
+          Class.new(ActiveRecord::Migration[version]) do
+            def change
+              create_table :testing_create_table_integer_id, :id => :integer do |t|
+                t.string "name"
+              end
+            end
+          end.migrate(:up)
+        end
+
+        klass.table_name = :testing_create_table_integer_id
+        expect(klass.attribute_names.sort).to eq(%w(id name))
+
+        klass.create!(:name => "abc")
+        expect(klass.first.id).to be > TestRecord::DEFAULT_RAILS_SEQUENCE_FACTOR
```

Comment:
I think the better way to do this is to check the sequence itself like on line 25.

Reply:
ok, I can check the sequence is > than the TestRecord::DEFAULT_RAILS_SEQUENCE_FACTOR... the point of the test was to show that requests for integer id columns would be ignored and bigserial would be used.